### ISSUE TITLE
TextDecoder and TextEncoder APIs

### ISSUE THREAD

**Issue by Redfire75369** [Redfire75369/spiderfire#26]
Spiderfire currently lacks a way to convert bytes into strings correctly and vice versa. In order to combat this, the `TextDecoder` and `TextEncoder` APIs should be implemented.

This will require use of `#[js_class]` for the primary APIs, and use of `encoding_rs` to support the many different encodings supported by the standard.

Implementation of `TextDecoderStream` and `TextEncoderStream` is not needed for the time being as #8 is required for that first.

### Resources
- [encoding_rs crate](https://crates.io/crates/encoding_rs)
- [MDN Documentation - TextDecoder](https://developer.mozilla.org/en-US/docs/Web/API/TextDecoder)
- [MDN Documentation - TextEncoder](https://developer.mozilla.org/en-US/docs/Web/API/TextEncoder)
- [Encoding Specification](https://encoding.spec.whatwg.org/)

**Comment by Redfire75369** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Add TextDecoder and TextEncoder](https://trello.com/c/sRVTYh0u/19-add-textdecoder-and-textencoder)

**Comment by Redfire75369** [comment]
This has been complete as of https://github.com/Redfire75369/spiderfire/commit/4708a2ea13ca953dd06fd9501221ed067510cedf. The remaining portion to be implemented relates to streams, which depends on #8 and will get their own issue if necessary.
